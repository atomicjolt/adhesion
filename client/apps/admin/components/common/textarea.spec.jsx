import React from 'react';
import { shallow } from 'enzyme';
import Textarea from './textarea';

describe('textarea', () => {
  let result;
  let props;
  let changed;

  beforeEach(() => {
    changed = false;
    props = {
      textareaProps: {
        id: 'IM AN ID',
        value: 'IM A VALUE',
        disabled: false,
        name: 'the name',
        placeholder: 'IMA PLACEHOLDER',
        maxLength: 1,
        minLength: 1,
        cols: 3,
        rows: 2,
        onChange: () => { changed = true; },
      },
      className: 'imaclass',
      labelText: 'IMA LABEL',
    };
    result = shallow(<Textarea {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the onChange event', () => {
    expect(changed).toBeFalsy();
    result.find('textarea').simulate('change');
    expect(changed).toBeTruthy();
  });
});
