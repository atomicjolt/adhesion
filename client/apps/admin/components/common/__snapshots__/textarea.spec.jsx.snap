// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textarea matches the snapshot 1`] = `ShallowWrapper {}`;
